src/Translation_unit.ml

```diff
@@ -114,94 +108,95 @@ let parse parse_ast (conf : Conf.t) ic =
     Location.warning_printer := warning_printer ;
     raise e
 
-(** Debug: dump internal ast representation to file. *)
-let dump xunit dir base suf ext ast =
-  if Conf.debug then (
-    let tmp = Filename.concat dir (base ^ suf ^ ext) in
-    let oc = Out_channel.create tmp in
-    xunit.printast (Caml.Format.formatter_of_out_channel oc) ast ;
-    Out_channel.close oc )
-
-type result =
-  | Ok
-  | Invalid_source of exn
-  | Unstable of int
-  | Ocamlformat_bug of exn
+type error =
+  | Invalid_source of {input_name: string; exn: exn}
+  | Unstable of
+      { input_name: string
+      ; iteration: int
+      ; prev: string
+      ; next: string }
+  | Ocamlformat_bug of {input_name: string; exn: exn}
   | User_error of string
 
 let ellipsis n msg =
   let msg = String.strip msg in
   if n > 0 && String.length msg > (n * 2) + 10 then
-    Format.sprintf "%s ... %s" (String.prefix msg n) (String.suffix msg n)
+    sprintf "%s ... %s" (String.prefix msg n) (String.suffix msg n)
   else msg
 
 let ellipsis_cmt = ellipsis 50
 
-let parse_print (XUnit xunit) (conf : Conf.t) ~input_name ~input_file ic
-    ofile =
+let make_dump xunit input_name dir =
   let dir =
-    match ofile with
-    | Some ofile -> Filename.dirname ofile
-    | None -> Filename.get_temp_dir_name ()
+    match dir with Some dir -> dir | None -> Filename.get_temp_dir_name ()
```

Comment:
Can be rewritten with `Option.value`

Reply:
This would force the evaluation of `Filename.get_temp_dir_name`. I guess it's ok in that case,

Reply:
@gpetiot, how do you feel about taking over that PR ? I don't have much time to spend on this. Feel free to push to the branch directly.